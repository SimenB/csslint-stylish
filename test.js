/* eslint-env jest */

import { CSSLint } from 'csslint';
import chalk from 'chalk';

import path from 'path';

import reporter from './stylish';

test('should report stuff', () => {
  const res = CSSLint.verify('.class {\n  color: red !important\n}\n');

  const report = reporter.startFormat() + reporter.formatResults(res, path.resolve('style.css')) + reporter.endFormat();

  expect(chalk.stripColor(report)).toMatchSnapshot();
});

test('should report with full path', () => {
  const res = CSSLint.verify('.class {\n  color: red !important\n}\n');

  const report = reporter.startFormat() +
    reporter.formatResults(res, path.resolve('style.css'), { absoluteFilePathsForFormatters: true }) +
    reporter.endFormat();

  expect(chalk.stripColor(report)).toMatchSnapshot();
});

test('should be able to be registered as formatter', () => {
  expect(CSSLint.hasFormat('stylish')).toEqual(false);

  CSSLint.addFormatter(reporter);

  expect(CSSLint.hasFormat('stylish')).toEqual(true);
});

test('should not report undefined output lines when no filename provided', () => {
  const res = CSSLint.verify('.class {\n  color: red !important\n}\n');

  const report = reporter.startFormat() + reporter.formatResults(res) + reporter.endFormat();

  expect(chalk.stripColor(report)).toMatchSnapshot();
});

test('should report filename provided', () => {
  const res = CSSLint.verify('.class {\n  color: red !important\n}\n');
  const filename = path.resolve('filenamestyle.css');
  const report = reporter.startFormat() +
    reporter.formatResults(res, filename, { absoluteFilePathsForFormatters: true }) +
    reporter.endFormat();

  expect(chalk.stripColor(report)).toMatchSnapshot();
});

test('should report no violations if there are none', () => {
  const res = CSSLint.verify('.class {\n  color: red;\n}\n');
  const filename = path.resolve('filenamestyle.css');
  const report = reporter.startFormat() + reporter.formatResults(res, filename) + reporter.endFormat();

  expect(chalk.stripColor(report)).toMatchSnapshot();
});

test('should report errors', () => {
  const res = CSSLint.verify('.class {\n  color: red !important\n}\n', { important: 2 });
  const filename = path.resolve('filenamestyle.css');
  const report = reporter.startFormat() + reporter.formatResults(res, filename) + reporter.endFormat();

  expect(chalk.stripColor(report)).toMatchSnapshot();
});

test('should report multiple warnings', () => {
  const res = CSSLint.verify('.class {\n  color: red !important;\ntext-size: 12px !important;\n}\n', { important: 1 });
  const filename = path.resolve('filenamestyle.css');
  const report = reporter.startFormat() + reporter.formatResults(res, filename) + reporter.endFormat();

  expect(chalk.stripColor(report)).toMatchSnapshot();
});

test('should report multiple errors', () => {
  const res = CSSLint.verify('.class {\n  color: red !important;\ntext-size: 12px !important;\n}\n', { important: 2 });
  const filename = path.resolve('filenamestyle.css');
  const report = reporter.startFormat() + reporter.formatResults(res, filename) + reporter.endFormat();

  expect(chalk.stripColor(report)).toMatchSnapshot();
});

test('should report rollups correctly', () => {
  const res = CSSLint.verify(
    '.class {\n  float: left;\n  float: left;\n  float: left;\n  float: left;\n  float: left;\n  float: left;' +
      '\n  float: left;\n  float: left;\n  float: left;\n  float: left;\n  float: left;\n}\n',
    { floats: 2 }
  );
  const filename = path.resolve('filenamestyle.css');
  const report = reporter.startFormat() + reporter.formatResults(res, filename) + reporter.endFormat();

  expect(chalk.stripColor(report)).toMatchSnapshot();
});
