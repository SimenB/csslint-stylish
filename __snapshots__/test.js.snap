// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not report undefined output lines when no filename provided 1`] = `
"
  line 2  char 3  Use of !important

    ⚠  1 warning
"
`;

exports[`should report errors 1`] = `
"
filenamestyle.css
  line 2  char 3  Use of !important

    ✖  1 error
"
`;

exports[`should report filename provided 1`] = `
"
/Users/simen/Development/csslint-stylish/filenamestyle.css
  line 2  char 3  Use of !important

    ⚠  1 warning
"
`;

exports[`should report multiple errors 1`] = `
"
filenamestyle.css
  line 2  char 3  Use of !important
  line 3  char 1  Use of !important

    ✖  2 errors
"
`;

exports[`should report multiple warnings 1`] = `
"
filenamestyle.css
  line 2  char 3  Use of !important
  line 3  char 1  Use of !important

    ⚠  2 warnings
"
`;

exports[`should report no violations if there are none 1`] = `
"

No violations"
`;

exports[`should report rollups correctly 1`] = `
"
filenamestyle.css
      Too many floats (11), you're probably using them for layout. Consider using a grid system instead.

    ⚠  1 warning
"
`;

exports[`should report stuff 1`] = `
"
style.css
  line 2  char 3  Use of !important

    ⚠  1 warning
"
`;

exports[`should report with full path 1`] = `
"
/Users/simen/Development/csslint-stylish/style.css
  line 2  char 3  Use of !important

    ⚠  1 warning
"
`;
